src/components/MainPage.js

```diff
@@ -0,0 +1,142 @@
+/* @flow */
+import React, { Component, Fragment } from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import Card from '@material-ui/core/Card';
+import CardActions from '@material-ui/core/CardActions';
+import CardContent from '@material-ui/core/CardContent';
+import CardMedia from '@material-ui/core/CardMedia';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+import LinearProgress from '@material-ui/core/LinearProgress';
+import 'App.css';
+
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as defaultMoviesReducer from 'modules/defaultMoviesReducer';
+import * as searchMoviesReducer from 'modules/searchMoviesReducer';
+
+const styles = () => ({
+  root: {
+    flexGrow: 1,
+  },
+  textField: {
+    width: 445,
+  },
+  card: {
+    maxWidth: 445,
+    margin: '20px auto',
+  },
+  media: {
+    height: 0,
+    paddingTop: '100%',
+  },
+});
+
+type Props = {
+  defaultAction: Object,
+  searchAction: Object,
+  data: Array<number | string>,
+  classes: Object,
+};
+
+type State = {
+  inputValue: number | string,
+  error: Object,
+};
+
+class MainPage extends Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {

```

Comment:
You have this data in store so you must just pass it from the redux store. No need to save the copy in the local state.